packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1584,6 +1600,72 @@ class CrudStoreImpl
   openCreateSearchIndexModal() {
     this.localAppRegistry.emit('open-create-search-index-modal');
   }
+
+  openBulkDeleteDialog() {
+    const PREVIEW_DOCS = 5;
+
+    this.setState({
+      bulkDelete: {
+        previews: this.state.docs?.slice(0, PREVIEW_DOCS) || [],
+        status: 'open',
+        affected: this.state.count || 0,
+      },
+    });
+  }
+
+  bulkDeleteInProgress() {
+    this.setState({
+      bulkDelete: {
+        ...this.state.bulkDelete,
+        status: 'in-progress',
+      },
+    }); // TODO: COMPASS-7328
+  }
+
+  bulkDeleteFailed(ex: Error) {
+    return ex; // TODO: COMPASS-7328
+  }
+
+  bulkDeleteSuccess() {
+    this.closeBulkDeleteDialog();
```

Comment:
the dialog is already closed when we call `runBulkDelete`.

Reply:
You are right, removing this.